### PR TITLE
feat: simplify interface

### PR DESCRIPTION
**Pull request by danbosnichill**
Make a bunch of changes:
- Remove compact functions.
- Remove `prepareForLogging` method.
- Use the same shadow traffic flag for non-experiment and experiment.  Simplifies the setup.
- Allows `deliver` with `onlyLog=true` to be able to call shadow traffic.
- Reduce boilerplate test code.
- Add toContents utility.

BREAKING CHANGE: This changes the external interface in multiple ways.

### CODE REVIEW

File: src/pager.test.ts [hunk 1]
```diff
@@ -10,19 +10,13 @@ describe('apply paging', () => {
   beforeEach(() => {
     insertions = [
       {
-        insertionId: 'uuid0-0',
-        requestId: 'uuid0-1',
-        viewId: 'uuid0-2',
+        contentId: '0',
```
Comment (danbosnichill) [inline src/pager.test.ts:13]: I removed insertionId from these tests because the prod code won't have insertionIds here.

File: src/client.ts [hunk 2]
```diff
@@ -272,78 +212,74 @@ export class PromotedClientImpl implements PromotedClient {
     // If defined, log the CohortMembership to Metrics API.
     let cohortMembershipToLog: CohortMembership | undefined = undefined;
 
+    // Set to empty to simplify some of the checks.
+    let requestInsertions = request.insertion ?? [];
```
Comment (scottmcmaster) [inline src/client.ts:216]: This code seems a little risky since after line 222 request.insertion may or may not be set and we can't be sure if subsequent code will try to use it (in the future if not now).

File: src/client.ts [hunk 3]
```diff
@@ -272,78 +212,74 @@ export class PromotedClientImpl implements PromotedClient {
...
-        fullInsertion,
-        deliveryRequest.insertionPageType,
-        requestToLog?.paging
-      );
+      responseInsertions = this.pager.applyPaging(requestInsertions, insertionPageType, request.paging);
+      addInsertionIds(responseInsertions, this.uuid);
+      const responseToLog = {
+        insertion: responseInsertions,
+      };
+      deliveryLogToLog = this.createSdkDeliveryLog(requestToLog, responseToLog);
```
Comment (scottmcmaster) [inline src/client.ts:272]: All the assignments to responseInsertions are hard to follow. I wonder if line 275 can go away if we just init it to [] above.

File: src/client.ts [hunk 4]
```diff
@@ -461,27 +381,9 @@ export class PromotedClientImpl implements PromotedClient {
...
-    // Clear the userInfo since we already copied it to the LogRequest.
-    delete copyRequest['userInfo'];
-
-    return copyRequest;
-  };
-
-  private preDeliveryFillInFields = (deliveryRequest: DeliveryRequest | MetricsRequest) => {
-    const { request } = deliveryRequest;
+  private fillInRequestFields = (request: Request): Request => {
+    // Create a copy so we do not modify the input.
```
Comment (scottmcmaster) [inline src/client.ts:385]: Since it's only a shallow copy it seems like we are modifying references to, say, timing. Not sure how much we care.

File: src/client.ts [hunk 5]
```diff
@@ -490,63 +392,21 @@ export class PromotedClientImpl implements PromotedClient {
...
-const compactPropertiesOnFullInsertion = (fullInsertion: Insertion, compactFn: PropertiesMapFn) => {
-  const { properties } = fullInsertion;
-  if (!properties) {
-    return fullInsertion;
-  }
-  const newProperties = compactFn(properties);
-  // If the same memory reference, do not copy.
-  if (newProperties === properties) {
-    return fullInsertion;
+const addInsertionIds = (responseInsertions: Insertion[], uuid: () => string) => {
```
Comment (scottmcmaster) [inline src/client.ts:402]: Could probably just collapse these two functions into a map.

File: src/map-response.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+ */
+export const toContents = <T extends HasInsertionId>(
+  responseInsertions: Insertion[],
+  contentLookup: Record<string, T>
+): T[] => {
+  const results: T[] = [];
+  responseInsertions.forEach((insertion) => {
+    const { contentId } = insertion;
+    if (!contentId) {
+      console.warn('Encounter missing Insertion.contentId');
```
Comment (scottmcmaster) [inline src/map-response.ts:19]: Isn't this an expected case if response insertions contains results from paging, though?

File: src/client.ts (hunk 2 above)
Comment (danbosnichill) [reply src/client.ts:216]: Not sure I understand the recommendation but I moved `request.insertion = requestInsertions;` out of the if conditions.

I could also pull the prepare steps into a separate.  LMK

File: src/client.ts (hunk 3 above)
Comment (danbosnichill) [reply src/client.ts:272]: I can remove one line by changing `responseInsertions` to not include undefined.

File: src/client.ts (hunk 4 above)
Comment (danbosnichill) [reply src/client.ts:385]: Good point.  I'll just do the copy approach.

File: src/client.ts (hunk 5 above)
Comment (danbosnichill) [reply src/client.ts:402]: Done.  I don't know about perf trade offs for this.

File: src/map-response.ts (hunk 6 above)
Comment (danbosnichill) [reply src/map-response.ts:19]: This one shouldn't happen.  It's the other warn that is more likely.  I'll remove the other warning.

File: src/client.ts (hunk 3 above)
Comment (danbosnichill) [reply src/client.ts:272]: LMK if this is what you were intending.

File: src/client.ts (hunk 2 above)
Comment (scottmcmaster) [reply src/client.ts:216]: It was a concern about "requestInsertions" and "request.insertion" being out-of-sync (including the latter possibly undefined), and if/when they get used down below. Whatever you think is ok.

File: src/client.ts (hunk 3 above)
Comment (scottmcmaster) [reply src/client.ts:272]: I think that's it, yeah, thanks.

**Review by scottmcmaster — APPROVED** [review]

File: src/client.ts (hunk 2 above)
Comment (danbosnichill) [reply src/client.ts:216]: Ah, okay.  I deleted `requestInsertions` and kept `request.insertion`.
